packages/core/lib/console-child.js

```diff
@@ -23,60 +18,33 @@ global.crypto = {
 const { network, config, url } = yargs(input[0]).argv;
 const detectedConfig = Config.detect({ network, config });
 
-function getConfiguredNetworkUrl(customConfig) {
-  const configuredNetworkOptions = {
-    host: customConfig.host || managedGanacheDefaultHost,
-    port: customConfig.port || managedGanacheDefaultPort
-  };
-  return `http://${configuredNetworkOptions.host}:${configuredNetworkOptions.port}/`;
-}
-
-let configuredNetwork;
-
-if (url) {
-  // Use "url" to configure network (implies not "develop")
-  configuredNetwork = {
-    network_id: "*",
-    url
-  };
-} else {
-  // Otherwise derive network settings
-  const customConfig = detectedConfig.networks[network] || {};
-
-  configuredNetwork = {
-    host: customConfig.host || managedGanacheDefaultHost,
-    port: customConfig.port || managedGanacheDefaultPort,
-    network_id: customConfig.network_id || managedGanacheDefaultNetworkId,
-    url: getConfiguredNetworkUrl(customConfig)
-  };
-}
+function main() {
+  const {
+    getCommand,
+    prepareOptions,
+    runCommand,
+    deriveConfigEnvironment
+  } = require("./command-utils");
+  const derivedConfig = deriveConfigEnvironment(detectedConfig, network, url);
+  const command = getCommand({ inputStrings, options: {}, noAliases: false });
+  const options = prepareOptions({
+    command,
+    inputStrings,
+    options: derivedConfig
+  });
 
-detectedConfig.networks[network] = {
-  ...configuredNetwork,
-  provider: function () {
-    return new Web3.providers.HttpProvider(configuredNetwork.url, {
-      keepAlive: false
+  runCommand(command, options)
+    .then(() => process.exit(0))
+    .catch(error => {
+      // Perform error handling ourselves.
+      if (error instanceof TruffleError) {
+        console.log(error.message);
+      } else {
+        // Bubble up all other unexpected errors.
+        console.log(error.stack || error.toString());
+      }
+      process.exit(1);
     });
-  }
-};
-
-const { getCommand, prepareOptions, runCommand } = require("./command-utils");
-const command = getCommand({ inputStrings, options: {}, noAliases: false });
-const options = prepareOptions({
-  command,
-  inputStrings,
-  options: detectedConfig
-});
+}
 
-runCommand(command, options)
-  .then(() => process.exit(0))
-  .catch(error => {
-    // Perform error handling ourselves.
-    if (error instanceof TruffleError) {
-      console.log(error.message);
-    } else {
-      // Bubble up all other unexpected errors.
-      console.log(error.stack || error.toString());
-    }
-    process.exit(1);
-  });
+main();
```

Comment:
Why did you need to put all of the above to be executed in a function here at the end?

Reply:
Oh! This is just for sometime in the future if we want this function to be exported for some reason.

Reply:
IIRC, this came about from refactoring the module to make unit testing easier. Why would it be easier? It's because this module is executed as a child-process and those tests are more complicated; factoring the logic to a different module simplifies writing unit tests. 

It could be extended to use a `require.main === module ? main() : throw new Error("console-child should not be imported")` check to prevent incorrect import, but I don't think it's necessary.